src/controller/CommissionerDiscoveryController.cpp

```diff
@@ -606,10 +627,23 @@ void CommissionerDiscoveryController::CommissioningSucceeded(uint16_t vendorId,
     mVendorId  = vendorId;
     mProductId = productId;
     mNodeId    = nodeId;
+
+    auto rotatingId = std::string{};
+    auto passcode = uint32_t{ 0 };
+
+    auto client = GetUDCClientState();
+    if (client != nullptr)
+    {
+        // Get rotating ID and cached (commissioner?) passcode to handle AccountLogin
+        rotatingId = getRotatingIdString(client);
+        // Q: Should we use mPasscode, client->GetCommissionerPasscode, or client->GetCachedCommissionerPasscode?
+        passcode = std::to_string(client->GetCachedCommissionerPasscode());
```

Comment:
@chrisdecenzo Would be very helpful to understand how to correctly use cached passcodes here.

Reply:
mPasscode is the value used for commissioning so no need to call client->GetCachedCommissionerPasscode(). In fact, sometimes the commissioner passcode will be wrong since the user manually entered the passcode.

Simply pass mPasscode in to CommissioningCompleted

Reply:
Updated to use cached mPasscode. See open questions about it, however.